gobblin-core/src/main/java/org/apache/gobblin/converter/avro/JsonRecordAvroSchemaToAvroConverter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.converter.avro;
+
+import java.util.List;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.gobblin.configuration.WorkUnitState;
+import org.apache.gobblin.converter.Converter;
+import org.apache.gobblin.converter.DataConversionException;
+import org.apache.gobblin.converter.SchemaConversionException;
+import org.apache.gobblin.converter.SingleRecordIterable;
+import org.apache.gobblin.converter.ToAvroConverterBase;
+import com.google.gson.JsonObject;
+
+
+/**
+ * {@link Converter} that takes an Avro schema in string format and corresponding {@link JsonObject} records and
+ * converts them to {@link GenericRecord} using the schema
+ */
+public class JsonRecordAvroSchemaToAvroConverter extends ToAvroConverterBase<String, JsonObject> {
+
+  /**
+   * Take an Avro schema in string format and parse it into a {@link Schema}
+   */
+  @Override
+  public Schema convertSchema(String schema, WorkUnitState workUnit) throws SchemaConversionException {
+    return new Schema.Parser().parse(schema);
```

Comment:
My thought was to use `StringSchemaInjector` as part of the converter chain to inject the Avro schema before this converter. But maybe it would be better to include that as part of this converter since you would always need an Avro schema from config to use this.